ui/src/store.js

```diff
@@ -0,0 +1,186 @@
+        if (fieldNumber === 0) {
+          inputPurse = purse;
+          if (inputPurse === outputPurse) {
+            outputPurse = null;
+          }
+        }
+        if (fieldNumber === 1) {
+          outputPurse = purse;
+          if (outputPurse === inputPurse) {
+            inputPurse = null;
+          }
+        }
+        return { ...state, inputPurse, outputPurse, freeVariable };
+      },
+    },
+    swapInputs(state) {
+      const { inputPurse, outputPurse, inputAmount, outputAmount } = state;
+      return {
+        ...state,
+        inputPurse: outputPurse,
+        outputPurse: inputPurse,
+        inputAmount: outputAmount,
+        outputAmount: inputAmount,
+      };
+    },
+    changeAmount: {
+      // Map positional arguments.
+      create: (amount, fieldNumber, freeVariable = null) => ({
+        amount,
+        fieldNumber,
+        freeVariable,
+      }),
+      reducer(state, { amount, fieldNumber, freeVariable }) {
+        let { inputAmount, outputAmount } = state;
+        if (fieldNumber === 0) {
+          inputAmount = amount;
+        }
+        if (fieldNumber === 1) {
+          outputAmount = amount;
+        }
+        return { ...state, inputAmount, outputAmount, freeVariable };
+      },
+    },
+    createOffer: {
+      // Map positional arguments.
+      create: (
+        instanceHandleBoardId,
+        installationHandleBoardId,
+        invitationDepositId,
+        inputAmount,
+        outputAmount,
+        inputPurse,
+        outputPurse,
+      ) => ({
+        instanceHandleBoardId,
+        installationHandleBoardId,
+        invitationDepositId,
+        inputAmount,
+        outputAmount,
+        inputPurse,
+        outputPurse,
+      }),
+      reducer(
+        state,
+        {
+          instanceHandleBoardId,
+          installationHandleBoardId,
+          invitationDepositId,
+          inputAmount,
+          outputAmount,
+          inputPurse,
+          outputPurse,
+        },
+      ) {
+        const offer = {
+          // JSONable ID for this offer.  Eventually this will be scoped to
+          // the current site.
+          id: Date.now(),
+
+          // TODO: get this from the invitation instead in the wallet. We
+          // don't want to trust the dapp on this.
+          instanceHandleBoardId,
+          installationHandleBoardId,
+
+          proposalTemplate: {
+            give: {
+              In: {
+                // The pursePetname identifies which purse we want to use
+                pursePetname: inputPurse.pursePetname,
+                value: inputAmount,
+              },
+            },
+            want: {
+              Out: {
+                pursePetname: outputPurse.pursePetname,
+                value: outputAmount,
+              },
+            },
+            exit: { onDemand: null },
+          },
+        };
+
+        // Create an invitation for the offer and on response, send the proposed
+        // offer to the wallet.
```

Comment:
Where is the "on response" behavior described here?

Reply:
I added a comment, pointing to `contexts/Application.jsx`.